src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonNull.cs

```diff
@@ -43,23 +43,34 @@ public sealed class JsonNull : JsonNode, IEquatable<JsonNull>
         /// </summary>
         /// <param name="other">The JSON null to compare against.</param>
         /// <returns><see langword="true"/></returns>
-        public bool Equals(JsonNull other) => true;
+        public bool Equals(JsonNull other) => !(other is null);
```

Comment:
What's the policy on `other is object`? 🤔

Reply:
After looking at the code again I noticed I missed some change to `Equals(object obj)` method

https://github.com/dotnet/runtime/blob/f2a9f007afac5a8a9f48c0edb2603b5830ad7fcc/src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonNull.cs#L33
Should have been
`public override bool Equals(object obj) => obj is JsonNull jsonNull && Equals(jsonNull);`

if this didn't cover your question would you mind rephrasing what you mean?

Reply:
I'm just wondering what the stance is on the `x is object` syntax is. The aspnet folks don't seem to love it, but it's equivalent to `!(x is null)`, see https://sharplab.io/#v2:CYLg1APgAgzABFATHAwgWAFAG9Nzw+AIwHtiAbOAMVIApjCArAUwGMAXOADwEpd8cM+IQgDscAIQ1OcAJYBnOADsArmTLcA3HD54Avju2D8sOCXJwAQgEMATnUasOPQ0IHDjY6fLj1m7DQa6hrpAA===

Reply:
I see, I did `!(x is null)` as that would follow how the other Json methods do it.

JsonString:

https://github.com/dotnet/runtime/blob/f2a9f007afac5a8a9f48c0edb2603b5830ad7fcc/src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonString.cs#L168

JsonBoolean:
https://github.com/dotnet/runtime/blob/f2a9f007afac5a8a9f48c0edb2603b5830ad7fcc/src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonBoolean.cs#L57

Reply:
Given they are quite similar (identical?), I would follow what other `JsonNode` types do, to remain internally consistent.

> After looking at the code again I noticed I missed some change to `Equals(object obj)` method

https://github.com/dotnet/runtime/blob/f2a9f007afac5a8a9f48c0edb2603b5830ad7fcc/src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonNull.cs#L33
> Should have been
`public override bool Equals(object obj) => obj is JsonNull jsonNull && Equals(jsonNull);`

Agreed. Please address that and add the appropriate test for that case :)